File: curie/sorting/PRO42747.java
package sorting;
import java.util.*;

public class PRO42747 {
    public int solution(int[] citations) {
        int answer = 0;
        Arrays.sort(citations);
        int len = citations.length;
        for(int i=0; i<len; i++){
            //오름차순 정렬되어있으므로 인덱스를 이용하여 h 이상 인용된 논문 개수 구할 수 있음
            //x번 인용된 논문이 y개 있는 경우, min(x,y)가 h가 됨
            int citeCount = citations[i];
            answer = Math.max(answer, Math.min(len-i, citeCount));
        }
        return answer;
    }
}

File: hackownw/Sorting/pro42747.cpp
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> citations){
    sort(citations.begin(),citations.end(),greater<int>());
    int length = citations.size();
    for (int i=0;i<length;i++){
        if (citations[i] <= i) return i;
    }
}

int main(){
    vector<int> citations;
    solution(citations);
    return 0;
}